### DISCUSSION TITLE
Can I define my own linter specific for my project?

### DISCUSSION THREAD

**Discussion by ag91** [clj-kondo/clj-kondo#1740] [Q&A] [answered]
Thanks for the amazing tool!

I am using [re-frame](https://github.com/day8/re-frame) and there are a bunch of pitfalls for new users. 
So, I was thinking to define my own linters to catch these up and help others avoid these problems.

Is there a way to define external linters that I can load in clj-kondo? Maybe using the configuration? 

Thanks again for this tool!

Andrea

**Comment by borkdude** [comment, marked as answer]
Hi @ag91 - yes this is possible using hooks: https://github.com/clj-kondo/clj-kondo/blob/master/doc/hooks.md

If you want a gentle introduction into hooks, you can walk through the clj-kondo hooks workshop given at ClojureD 2022:

https://github.com/clj-kondo/hooks-workshop-clojured-2022

**Reply by ag91** [reply]
thanks I am going to try!
